src/test/java/com/algolia/search/integration/insights/InsightsTest.java

```diff
@@ -82,5 +70,7 @@ void testInsights() throws ExecutionException, InterruptedException {
             Arrays.asList("1", "2"),
             search2.getQueryID())
         .get();
+
+    insightsClient.close();
```

Comment:
Maybe we could use a try with ressources to ensure `insightsClient`is closed in case of test failure?

Reply:
Agreed. When I read your comment I first had in mind to suggest putting that in another review, but it actually makes sense to have it there: prior to the patch, an error in the very test wouldn't have prevented the closing of the client, we must keep this guarantee.

Reply:
done